sdk/identity/src/token_credentials/azureauth_cli_credentials.rs

```diff
@@ -0,0 +1,182 @@
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct CliTokenResponse {
+    pub token: AccessToken,
+    #[serde(with = "unix_date_string")]
+    pub expiration_date: OffsetDateTime,
+}
+
+/// Authentication Mode
+///
+/// Note: While the azureauth CLI supports devicecode, users wishing to use
+/// devicecode should use `azure_identity::device_code_flow`
+#[derive(Debug, Clone, Copy)]
+pub enum AzureauthCliMode {
+    All,
+    IntegratedWindowsAuth,
+    Broker,
+    Web,
+}
+
+/// Enables authentication to Azure Active Directory using Azure CLI to obtain an access token.
+pub struct AzureauthCliCredential {
+    tenant_id: String,
+    client_id: ClientId,
+    modes: Vec<AzureauthCliMode>,
+}
+
+impl AzureauthCliCredential {
+    /// Create a new `AzureCliCredential`
+    pub fn new<T, C>(tenant_id: T, client_id: C) -> Self
+    where
+        T: Into<String>,
+        C: Into<ClientId>,
+    {
+        Self {
+            tenant_id: tenant_id.into(),
+            client_id: client_id.into(),
+            modes: Vec::new(),
+        }
+    }
+
+    pub fn add_mode(mut self, mode: AzureauthCliMode) -> Self {
+        self.modes.push(mode);
+        self
+    }
+
+    pub fn with_modes(mut self, modes: Vec<AzureauthCliMode>) -> Self {
+        self.modes = modes;
+        self
+    }
+
+    fn get_access_token(&self, resource: &str) -> azure_core::Result<CliTokenResponse> {
+        // try using azureauth.exe first, such that azureauth through WSL is
+        // used first if possible.
+        let cmd_name = if Command::new("azureauth.exe")
+            .arg("--version")
+            .output()
+            .map(|x| x.status.success())
+            .unwrap_or(false)
+        {
+            "azureauth.exe"
+        } else {
+            "azureauth"
+        };
+
+        let mut cmd = Command::new(cmd_name);
+        cmd.args([
+            "--resource",
+            resource,
+            "--client",
+            self.client_id.as_str(),
+            "--tenant",
+            self.tenant_id.as_str(),
+            "--output",
+            "json",
+        ]);
+
+        for mode in &self.modes {
+            cmd.arg("--mode");
+            match mode {
+                AzureauthCliMode::All => cmd.arg("all"),
+                AzureauthCliMode::IntegratedWindowsAuth => cmd.arg("iwa"),
+                AzureauthCliMode::Broker => cmd.arg("broker"),
+                AzureauthCliMode::Web => cmd.arg("web"),
+            };
+        }
+
+        let result = cmd.output();
+
+        let output = result.map_err(|e| match e.kind() {
+            std::io::ErrorKind::NotFound => {
+                Error::message(ErrorKind::Other, "azureauth CLI not installed")
+            }
+            error_kind => Error::with_message(ErrorKind::Other, || {
+                format!("Unknown error of kind: {error_kind:?}")
+            }),
+        })?;
+
+        if !output.status.success() {
+            let output = String::from_utf8_lossy(&output.stderr);
```

Comment:
Does `Command` already deal with the Unicode->UTF8 conversion? Output on Windows is not UTF8 by default unless explicit set as the codepage - even for PowerShell (had to deal with this for another app where proper UTF8 encoding was critical).

Reply:
I don't think it does.  I tested the output empirically to validate it worked as expected on error.  Additionally, this is how the az-cli integration works.

Reply:
AzureAuth explicitly sets it's output encoding as UTF-8.